### PR TITLE
feat(ip): add intellectual properties api domain

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

Add a new api domain, Intellectual properties
Feature-flagged.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by lodmfjord — APPROVED** [review]
Change to codeowners LGTM.

File: libs/api/domains/intellectual-properties/src/lib/intellectualProperties.service.ts
Comment (thordurhhh) [inline libs/api/domains/intellectual-properties/src/lib/intellectualProperties.service.ts:305]: what a comment 😂

File: libs/api/domains/rights-portal/src/lib/therapy/therapy.resolver.ts
Comment (thordurhhh) [inline libs/api/domains/rights-portal/src/lib/therapy/therapy.resolver.ts:19]: Remove?

**Review by thordurhhh — APPROVED** [review]
Niiice 🙌

File: libs/api/domains/intellectual-properties/src/lib/intellectualProperties.resolver.ts
Comment (saevarma) [inline libs/api/domains/intellectual-properties/src/lib/intellectualProperties.resolver.ts:40]: ~ FYI you can also just apply the `@Scopes` decorator on the class level.

File: libs/api/domains/intellectual-properties/src/lib/models/person.model.ts
Comment (saevarma) [inline libs/api/domains/intellectual-properties/src/lib/models/person.model.ts:40]: ~ The monorepo favors `nationalId` over `ssn`.

**Review by saevarma — APPROVED** [review]
Approved with optional comments but removed the `automerge` label so you can review them before this is merged.

File: libs/api/domains/intellectual-properties/src/lib/models/person.model.ts
Comment (thorkellmani) [reply libs/api/domains/intellectual-properties/src/lib/models/person.model.ts:40]: ah oops, will fix

**Pull request merged**
